src/testrender/background.h

```diff
@@ -115,14 +239,20 @@ struct Background {
         return make_Vec3(sin_phi * ct, sin_phi * st, cos_phi);
     }
 
-    static float sample_cdf(const float* data, unsigned int n, float x,
-                            unsigned int* idx, float* pdf)
+    static OSL_HOSTDEVICE float sample_cdf(const float* data, unsigned int n,
+                                           float x, unsigned int* idx,
+                                           float* pdf)
     {
-        OSL_DASSERT(x >= 0);
-        OSL_DASSERT(x < 1);
+        OSL_DASSERT(x >= 0.0f);
+        OSL_DASSERT(x < 1.0f);
+#ifndef __CUDACC__
         *idx = std::upper_bound(data, data + n, x) - data;
```

Comment:
I don't see any reason to keep using the `std::` version on CPU (the function is simple enough and I'm not aware of any c++ standard library that does anything fancier than your implementation).

Reply:
I'll leave it as CUDA-only for now to possibly do some linear search experiments, to see if that is a win on the GPU.